src/lib/Libtpp/tpp_router.c

```diff
@@ -288,6 +286,7 @@ send_leaves_to_router(tpp_router_t *parent, tpp_router_t *target)
 		free(lf_data->addrs);
 		free(lf_data);
 	}
+	free(pkey);
```

Comment:
why did you move this? pkey is not really used in the while block right?

Reply:
I moved it since it could cause a double free before.  It was freed on line 274.  On line 284 we goto err.  In err we free pkey.  I just moved pkey down past the goto err so we didn't have a chance to double free.  Aren't static analyzers grand?  They find the weirdest esoteric paths through the code that might cause a problem :)

Reply:
ah, ok, that makes sense. Just to be safe, can we set pkey to NULL after the free() ? Just in case somebody changes this code in the future.
Static analyzers are grand indeed.

Reply:
I don't really think it is necessary.  We return immediately after freeing pkey.  If someone modifies the code, I think it is pretty obvious that they shouldn't use pkey.  It is freed right above a return.